src/main/java/com/nvd/footballmanager/service/MatchService.java

```diff
@@ -1,4 +1,171 @@
+                        MemberRepository memberRepository) {
+        super(matchRepository, matchMapper);
+        this.matchRepository = matchRepository;
+        this.matchMapper = matchMapper;
+        this.memberService = memberService;
+        this.userService = userService;
+        this.teamRepository = teamRepository;
+        this.mailService = mailService;
+        this.notificationService = notificationService;
+        this.memberRepository = memberRepository;
+    }
+
+    @Override
+    @Transactional
+    public MatchDTO create(MatchDTO dto) {
+        Optional<Member> manager = memberService.currentUserMemberInTeam(dto.getTeam1Id());
+        if (!(manager.isPresent() && manager.get().getRole().equals(MemberRole.MANAGER)))
+            throw new AccessDeniedException(Constants.NOT_MANAGER_PERMISSION);
+
+        Team team2 = teamRepository.findById(dto.getTeam2Id())
+                .orElseThrow(() -> new EntityNotFoundException(Constants.ENTITY_NOT_FOUND));
+
+        Match match = matchMapper.convertToEntity(dto);
+        match.setTeam1(manager.get().getTeam());
+        match.setTeam2(team2);
+        Set<Member> teamMembers = new HashSet<>(memberRepository.findAllByTeamId(dto.getTeam1Id()));
+        match.setMembers(teamMembers);
+
+        // manager send noti to team members
+        NotiSendRequest notiToTeamMembers = new NotiSendRequest();
+        notiToTeamMembers.setTeamId(manager.get().getTeam().getId());
+        notiToTeamMembers.setTitle("New match created");
+        notiToTeamMembers.setContent("New match has been created between teams: " +
+                manager.get().getTeam().getName() + " and " + team2.getName() + " at " + match.getVenue()
+                + " on " + match.getTime());
+        notificationService.managerSendNotificationToTeam(notiToTeamMembers);
+
+        Match savedMatch = matchRepository.save(match);
+        return matchMapper.convertToDTO(savedMatch);
+    }
+
+    @Override
+    @Transactional
+    public MatchDTO update(UUID id, MatchDTO dto) {
+
+        Optional<Member> manager = memberService.currentUserMemberInTeam(dto.getTeam1Id());
+        if (!(manager.isPresent() && manager.get().getRole().equals(MemberRole.MANAGER)))
+            throw new AccessDeniedException(Constants.NOT_MANAGER_PERMISSION);
+
+        Team team2 = teamRepository.findById(dto.getTeam2Id())
+                .orElseThrow(() -> new EntityNotFoundException(Constants.ENTITY_NOT_FOUND));
+
+        Match match = matchRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Constants.ENTITY_NOT_FOUND));
+
+        // send updated noti to team members
+        NotiSendRequest notiToTeamMembers = new NotiSendRequest();
+        notiToTeamMembers.setTeamId(manager.get().getTeam().getId());
+        notiToTeamMembers.setTitle("Manager updated match information.");
+        StringBuilder content = new StringBuilder("Match has been updated between teams: " +
```

Comment:
Suggest Using Constrant instead of Plain Text

Reply:
dạ